### DISCUSSION TITLE
Geolocator.RequestAccessAsync Method

### DISCUSSION THREAD

**Discussion by sigmarsson** [unoplatform/uno#7708] [General]
###  Important

Starting in Windows 10, call the RequestAccessAsync method before accessing the user’s location. At that time, your app must be in the foreground and RequestAccessAsync must be called from the UI thread. Your app can then handle the no-permissions case without throwing an exception.

[MS doc suggests](https://docs.microsoft.com/en-us/uwp/api/windows.devices.geolocation.geolocator.getgeopositionasync?f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(Windows.Devices.Geolocation.Geolocator.GetGeopositionAsync);k(SolutionItemsProject);k(DevLang-csharp)%26rd%3Dtrue&view=winrt-22000) to invoke this method on UI thread before accessing geo location, like

`                var location = await new Geolocator().GetGeopositionAsync();`

Will you add this method to your `Uno.WinUI.Geolocator` class ?

**Comment by pkar70** [comment]
But why `Uno.WinUI.Geolocator` and not `Windows.Devices.Geolocation.Geolocator` ?
Windows.Devices.Geolocation.Geolocator.RequestAccessAsync  is implemented in Uno,

**Reply by sigmarsson** [reply]
just for brevity. you are right. it is already implemented. I wonder why earlier IntelliSense has failed to show me the method.

**Comment by sigmarsson** [comment]
I attempt to invoke this API in a console application and face this error. Aren't Uno APIs intended to consume in console applications ? One of my unit test requires a peer process to run whilst unit testing.

The assembly, which is being tested is referencing the package `Uno.WinUI 4.0.11`

![image](https://user-images.githubusercontent.com/43340959/147932199-4e1511e7-f63d-4f68-8f51-358944d22b0f.png)

**Reply by pkar70** [reply]
Geolocator.RequestAccessAsync() requires UI thread (as it should be able to display dialogs for user interaction). Maybe console app is not allowed here (in UWP, so also in any other platforms).

**Reply by jeromelaban** [reply]
@sigmarsson which platform are you targeting? If it is Gtk+Skia or WPF+Skia, `GeoLocator` is not implemented yet for these.

**Reply by sigmarsson** [reply]
@pkar70 okay. 👍 
@jeromelaban it's Windows 10/11 desktop only for now.
Thank you.
